lib/ansible/modules/database/postgresql/pg_hba.py

```diff
@@ -0,0 +1,597 @@
+        # (at least that are handled by this pg_hba) since this migth also be groups with multiple
+        # users, this migth be totally off, but at least it is some info...
+        self.users = set(['postgres'])
+
+        self.read()
+
+    def unchanged(self):
+        '''
+        This method resets self.diff to a empty default
+        '''
+        self.diff = {'before': {'file': self.pg_hba_file, 'pg_hba': []},
+                     'after': {'file': self.pg_hba_file, 'pg_hba': []}}
+
+    def read(self):
+        '''
+        Read in the pg_hba from the system
+        '''
+        self.rules = {}
+        self.comment = []
+        # read the pg_hbafile
+        try:
+            file = open(self.pg_hba_file, 'r')
+            for line in file:
+                line = line.strip()
+                # uncomment
+                if '#' in line:
+                    line, comment = line.split('#', 1)
+                    self.comment.append('#' + comment)
+                rule = self.line_to_rule(line)
+                if rule:
+                    self.add_rule(rule)
+            file.close()
+            self.unchanged()
+        except IOError:
+            if not self.create:
+                raise PgHbaError("pg_hba file '{0}' doesn't exist. "
+                                 "Use create option to autocreate.".format(self.pg_hba_file))
+
+    def line_to_rule(self, line):
+        '''
+        split into sid, home, enabled
+        '''
+        if WHITESPACES_RE.sub('', line) == '':
+            # empty line. skip this one...
+            return None
+        cols = WHITESPACES_RE.split(line)
+        if len(cols) < 4:
+            msg = "File {0} has a rule with too few columns: {1}."
+            raise PgHbaError(msg.format(self.pg_hba_file, line))
+        if cols[0] not in PG_HBA_TYPES:
+            msg = "File {0} contains an rule of unknown type: {1}."
+            raise PgHbaError(msg.format(self.pg_hba_file, line))
+        if cols[0] == 'local':
+            if cols[3] not in PG_HBA_METHODS:
+                raise PgHbaError("File {0} contains an rule of 'local' type where 4th column '{1}' "
+                                 "isn't a valid auth-method.".format(self.pg_hba_file, cols[3]))
+            cols.insert(3, None)
+            cols.insert(3, None)
+        else:
+            if len(cols) < 6:
+                cols.insert(4, None)
+            elif cols[5] not in PG_HBA_METHODS:
+                cols.insert(4, None)
+            if len(cols) < 7:
+                cols.insert(7, None)
+            if cols[5] not in PG_HBA_METHODS:
+                raise PgHbaError("File {0} contains an rule '{1}' "
+                                 "that has no valid method.".format(self.pg_hba_file, line))
+        rule = dict(zip(PG_HBA_HDR, cols[:7]))
+        clean_empty_rule_keys(rule)
+        rule['line'] = line
+        return rule
+
+    def rule2weight(self, rule):
```

Comment:
Using `order: sdu`, the file below was generated:
```
local   all             postgres                                peer
host    all             all             127.0.0.1/32            md5
local   all             all                                     peer
host    all             all             ::1/128                 md5
```
should not that be
```
local   all             postgres                                peer
local   all             all                                     peer
host    all             all             127.0.0.1/32            md5
host    all             all             ::1/128                 md5
``` 
?

Reply:
Although this is technically not important, I fixed this.
I also added some better testcases to pg_hba_test_ips to better test this.